src/syft/core/node/common/node.py

```diff
@@ -286,14 +286,7 @@ def known_child_nodes(self) -> List[Address]:
         if sy.VERBOSE:
             print(f"> {self.pprint} Getting known Children Nodes")
         if self.child_type_client_type is not None:
-            nodes = []
-            for key in self.store.keys():
-                value = self.store[key]
-                nodes.append(value)
-
-            # TODO: Make this work again
-            # nodes = self.store.get_objects_of_type(obj_type=self.child_type_client_type)
```

Comment:
I'm not sure why this was commented out previously but in theory the child types shouldn't be Address types they should be self.child_type_client_type types. Meaning DomainNode will have type DeviceClient and DeviceNode will have child type VMClient.

Since they all subclass from Address this PR probably passes tests but in the long run it could cause issues. That being said, we also need to put a TODO where we explicitly say that a client is a child as this is a special case. But this requires us to finish the metadatastore first. So, for now let's just make sure we're using the narrowest type we can.

The main risk with the current functionality is that if a DomainNode has a client to another DomainNode it will think that that client is its child. I *think* our security implementation is still robust against this but it could certainly create strange behavior.

Reply:
That's what I was thinking!
But I think that we now only have `Adresses` in the store and not the client types anymore (I found a comment somewhere telling that). So the information about the client's type seems lost... Or maybe I missed something?

Reply:
Do you think `Address` could have a method/property to tell of which type it is (Network, Domain, ...)?

Reply:
I think we should simply store the Client object in the store. It's designed to cache all the information needed to be able to communicate with a specific location. Without all the other information, storing the address has no use (even if we know the type)

Reply:
The Clients are stored in a `in_memory_client_registry` attribute on Nodes, but it's advertised as a temporary hack...
According to what I understood, Clients are not stored in the main store because they can't be serialized, am I right?

Reply:
Yes, to send a message you must have a python reference to a client / connection object which you can call "send_immediate_msg_with_reply" (and its other variations) on, however the serialized version of a client will not unserialize back to a full graph of the connection / route. We could remove this hack by running a solo version of duet with a flask webserver and post messages back to localhost/{uid} so that no actual reference to a client running in memory is required to dispatch, as all nodes could be listening to their address url stub on localhost? I am interested to know how we should handle this more long term.